tests/test_integration/test_toolkit_client/test_extended_files.py

```diff
@@ -47,3 +49,55 @@ def test_set_pending_instance_id(self, dev_cluster_client: ToolkitClient, dev_sp
             if created_dm is not None:
                 # This will delete the CogniteFile and the asset-centric file
                 client.data_modeling.instances.delete(cognite_file.as_id())
+
+    def test_unlink_instance_ids(self, toolkit_client_with_pending_ids: ToolkitClient, toolkit_space: Space) -> None:
+        client = toolkit_client_with_pending_ids
+        space = toolkit_space.space
+        metadata = FileMetadataWrite(
+            external_id="file_toolkit_integration_test_unlink",
+            name="Toolkit Integration Test Unlink",
+            mime_type="text/plain",
+        )
+        cognite_file = CogniteFileApply(
+            space=space,
+            external_id=metadata.external_id,
+            name="Toolkit Integration Test Unlink",
+        )
+        content = b"Hello, this is a test file's content."
+        created: FileMetadata | None = None
+        created_dm: NodeApplyResultList | None = None
+        try:
+            created, _ = client.files.create(metadata)
+            client.files.upload_content_bytes(content, external_id=created.external_id)
+
+            updated = client.files.set_pending_ids(cognite_file.as_id(), id=created.id)
+
+            assert updated.pending_instance_id == cognite_file.as_id()
+
+            created_dm = client.data_modeling.instances.apply(cognite_file).nodes
+
+            retrieved_ts: FileMetadata | None = None
+            for _ in range(10):
+                retrieved_ts = client.files.retrieve(instance_id=cognite_file.as_id())
+                if retrieved_ts is not None:
+                    break
+                time.sleep(1)  # Wait for the syncer to update the file metadata
+
+            assert retrieved_ts is not None, "File was not linked to instance within timeout"
+            assert retrieved_ts.id == created.id
+
+            unlinked = client.files.unlink_instance_ids(id=created.id)
+            assert unlinked.id == created.id
+
+            client.data_modeling.instances.delete(cognite_file.as_id())
+            created_dm = None
+
+            # Still existing asset-centric file.
+            retrieved_ts = client.files.retrieve(external_id=metadata.external_id)
+            assert retrieved_ts is not None
+            assert retrieved_ts.id == created.id
+        finally:
+            if created is not None and created_dm is None:
+                client.files.delete(external_id=metadata.external_id, ignore_unknown_ids=True)
```

Comment:
The files API does not allow deleting files if `pendingInstanceId` is set. This is different from the time series API.

So if the test fails to create the CogniteFile for some reason (which is already a valid reason to fail the test imho.), then you will not be able to delete it until you have unlinked it. So this line will fail.

That said, I hope they change this behavior in the files API, as I don't see a good reason for this restriction.